src/core/indexer/tables.rs

```diff
@@ -0,0 +1,454 @@
+    Ed448_Sig,
+    Ed25519_Big_Sig,
+    ECDSA_256k1_Big_Sig,
+    Ed448_Big_Sig,
+}
+
+impl BothSigCodex {
+    pub(crate) fn code(&self) -> &'static str {
+        match self {
+            BothSigCodex::Ed25519_Sig => "A", // Ed25519 sig appears same in both lists if any.
+            BothSigCodex::ECDSA_256k1_Sig => "C", // ECDSA secp256k1 sig appears same in both lists if any.
+            BothSigCodex::Ed448_Sig => "0A",      // Ed448 signature appears in both lists.
+            BothSigCodex::Ed25519_Big_Sig => "2A", // Ed25519 sig appears in both listsy.
+            BothSigCodex::ECDSA_256k1_Big_Sig => "2C", // ECDSA secp256k1 sig appears in both lists.
+            BothSigCodex::Ed448_Big_Sig => "3A",  // Ed448 signature appears in both lists.
+        }
+    }
+
+    pub(crate) fn from_code(code: &str) -> Result<Self> {
+        Ok(match code {
+            "A" => BothSigCodex::Ed25519_Sig,
+            "C" => BothSigCodex::ECDSA_256k1_Sig,
+            "0A" => BothSigCodex::Ed448_Sig,
+            "2A" => BothSigCodex::Ed25519_Big_Sig,
+            "2C" => BothSigCodex::ECDSA_256k1_Big_Sig,
+            "3A" => BothSigCodex::Ed448_Big_Sig,
+            _ => return Err(Box::new(Error::UnexpectedCode(code.to_string()))),
+        })
+    }
+
+    pub(crate) fn has_code(code: &str) -> bool {
+        vec![
+            BothSigCodex::Ed25519_Sig.code(),
+            BothSigCodex::ECDSA_256k1_Sig.code(),
+            BothSigCodex::Ed448_Sig.code(),
+            BothSigCodex::Ed25519_Big_Sig.code(),
+            BothSigCodex::ECDSA_256k1_Big_Sig.code(),
+            BothSigCodex::Ed448_Big_Sig.code(),
+        ]
+        .contains(&code)
+    }
+}
+
+/// Sizes table maps hs chars of code to Sizage (hs), ss, os, fs, ls)
+/// where hs is hard size, ss is soft size, os is other index size,
+/// and fs is full size, ls is lead size.
+/// where ss includes os, so main index size ms = ss - os
+/// soft size, ss, should always be  > 0 for Indexer
+#[derive(Debug, PartialEq)]
+pub struct Sizage {
+    pub hs: u32,
+    pub ss: u32,
+    pub os: u32,
+    pub ls: u32,
+    pub fs: u32,
+}
+
+impl Sizage {
+    pub fn new(hs: u32, ss: u32, os: u32, fs: u32, ls: u32) -> Sizage {
+        Self { hs, ss, os, ls, fs }
+    }
+}
+
+pub(crate) fn sizage(s: &str) -> Result<Sizage> {
+    match s {
+        "A" => Ok(Sizage::new(1, 1, 0, 88, 0)),
+        "B" => Ok(Sizage::new(1, 1, 0, 88, 0)),
+        "C" => Ok(Sizage::new(1, 1, 0, 88, 0)),
+        "D" => Ok(Sizage::new(1, 1, 0, 88, 0)),
+        "0A" => Ok(Sizage::new(2, 2, 1, 156, 0)),
+        "0B" => Ok(Sizage::new(2, 2, 1, 156, 0)),
+        "2A" => Ok(Sizage::new(2, 4, 2, 92, 0)),
+        "2B" => Ok(Sizage::new(2, 4, 2, 92, 0)),
+        "2C" => Ok(Sizage::new(2, 4, 2, 92, 0)),
+        "2D" => Ok(Sizage::new(2, 4, 2, 92, 0)),
+
+        "3A" => Ok(Sizage::new(2, 6, 3, 160, 0)),
+        "3B" => Ok(Sizage::new(2, 6, 3, 160, 0)),
+
+        "0z" => Ok(Sizage::new(2, 2, 0, u32::MAX, 0)),
```

Comment:
I took the approach of setting to 0 instead of u32::MAX, but I think this is actually nicer. You'll need to update a condition or two (where you check that `sizage_.fs == 0`) you copied in `indexer/mod.rs` I believe.

Reply:
I need to go back and check the python tbh, there are checks for `0` and `None`, `u32::MAX` is in place of `None`